dvc/stage/__init__.py

```diff
@@ -87,18 +87,6 @@ def create_stage(cls, repo, path, external=False, **kwargs):
     check_circular_dependency(stage)
     check_duplicated_arguments(stage)
 
-    if stage and stage.dvcfile.exists():
-        has_persist_outs = any(out.persist for out in stage.outs)
-        ignore_run_cache = (
-            not kwargs.get("run_cache", True) or has_persist_outs
-        )
-        if has_persist_outs:
-            logger.warning("Build cache is ignored when persisting outputs.")
-
-        if not ignore_run_cache and stage.can_be_skipped:
-            logger.info("Stage is cached, skipping")
-            return None
-
```

Comment:
Minimal changes to get this out of the way from create_stage. `can_be_skipped` and `is_cached` relations still raise questions. These days the connection to actual run_cache is more apparent, but in this functionality we mixup run_cache and avoiding to write the same contents to the dvcfile. Those need to be separate.